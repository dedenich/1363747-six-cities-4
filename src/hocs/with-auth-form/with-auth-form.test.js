import React from 'react';
import renderer from 'react-test-renderer';
import withAuthForm from "./with-auth-form.jsx";
import {Provider} from 'react-redux';
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

const MockComponent = () => <div></div>;

const MockComponentWrapped = withAuthForm(MockComponent);

it(`withAuthForm is rendered correctly`, () => {
  const store = mockStore({
    authorizationStatus: `NO_AUTH`,
    authInfo: null,
  });
  const tree = renderer
  .create(
      <Provider store={store}>
        <MockComponentWrapped/>
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
