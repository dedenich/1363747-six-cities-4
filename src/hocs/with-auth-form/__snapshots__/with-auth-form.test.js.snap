// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAuthForm is rendered correctly 1`] = `<div />`;
